segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/S3ClientWrapper.java

```diff
@@ -0,0 +1,264 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host;
+
+import com.emc.object.Range;
+import com.emc.object.s3.S3Config;
+import com.emc.object.s3.S3Exception;
+import com.emc.object.s3.S3ObjectMetadata;
+import com.emc.object.s3.bean.AccessControlList;
+import com.emc.object.s3.bean.CompleteMultipartUploadResult;
+import com.emc.object.s3.bean.CopyPartResult;
+import com.emc.object.s3.bean.DeleteObjectsResult;
+import com.emc.object.s3.bean.GetObjectResult;
+import com.emc.object.s3.bean.ListObjectsResult;
+import com.emc.object.s3.bean.PutObjectResult;
+import com.emc.object.s3.bean.S3Object;
+import com.emc.object.s3.jersey.S3JerseyClient;
+import com.emc.object.s3.request.CompleteMultipartUploadRequest;
+import com.emc.object.s3.request.CopyPartRequest;
+import com.emc.object.s3.request.DeleteObjectsRequest;
+import com.emc.object.s3.request.PutObjectRequest;
+import com.emc.object.s3.request.SetObjectAclRequest;
+import io.pravega.common.io.StreamHelpers;
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import io.pravega.segmentstore.storage.impl.extendeds3.AclSize;
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import lombok.SneakyThrows;
+import lombok.Synchronized;
+
+/**
+ * Client wrapper for S3JerseyClient.
+ */
+class S3ClientWrapper extends S3JerseyClient {
+    private final ConcurrentMap<String, AclSize> aclMap;
+    private final String baseDir;
+    private final ConcurrentMap<String, ConcurrentMap<Integer, CopyPartRequest>> multipartUploads = new ConcurrentHashMap<>();
+
+    public S3ClientWrapper(S3Config s3Config, ConcurrentMap<String, AclSize> aclMap, String baseDir) {
+        super(s3Config);
+        this.aclMap = aclMap;
+        this.baseDir = baseDir;
+    }
+
+    @Override
+    public DeleteObjectsResult deleteObjects(DeleteObjectsRequest request) {
+        return new DeleteObjectsResult();
+    }
+
+    @Synchronized
+    @Override
+    public PutObjectResult putObject(PutObjectRequest request) {
+
+        if (request.getObjectMetadata() != null) {
+            request.setObjectMetadata(null);
+        }
+        try {
+            Path path = Paths.get(this.baseDir, request.getBucketName(), request.getKey());
+            Files.createDirectories(path.getParent());
+            Files.createFile(path);
+        } catch (IOException e) {
+            throw new S3Exception(e.getMessage(), 0);
```

Comment:
Not according to the Javadoc: http://emcecs.github.io/ecs-object-client-java/latest/javadoc/com/emc/object/s3/S3Exception.html

Also the source code: https://github.com/EMCECS/ecs-object-client-java/blob/master/src/main/java/com/emc/object/s3/S3Exception.java